lib/bridge-client/blacklist.js

```diff
@@ -21,10 +21,36 @@ function Blacklist(options) {
   this.blacklistKey = '.blacklist';
   this._store = options.store;
   this._logger = options.logger;
+  // creating queue to prevent race conditions;
+  // only write one json object at a time
+  this._next = [];
+  this._writing = [];
 }
 
 Blacklist.TTL = 86400000;
 
+Blacklist.prototype._queueForWriting = function (cb) {
+  cb = cb || function() {};
+
+  if (this._writing.length > 0) {
+    return this._next.push(cb);
+  }
+  this._writing.push(cb);
+  this._saveToStore(this._doneWriting.bind(this));
+};
+
+Blacklist.prototype._doneWriting = function (e) {
+  this._writing.map(function(cb) {
```

Comment:
Not a big deal, but we should probably do this as a `for` loop. `.map` creates a copy of the array that we don't need.